src/components/Button/ButtonDropdown.react.js

```diff
@@ -0,0 +1,68 @@
+// @flow
+
+import * as React from "react";
+import { Button, Dropdown } from "../";
+import { Manager, Reference } from "react-popper";
+import type { Props as ButtonProps } from "../Button/Button.react";
+import type { Props as DropdownMenuProps } from "../Dropdown/DropdownMenu.react";
+
+import type { ReferenceChildrenProps } from "react-popper";
+
+type Props = ButtonProps & {|
+  +value?: string,
+  +children?: React.Node,
+  +dropdownProps?: DropdownMenuProps,
+|};
+
+type State = {
+  isOpen: boolean,
+};
+
+class ButtonDropdown extends React.Component<Props, State> {
+  state = { isOpen: false };
+
+  _handleButtonOnClick = (e: SyntheticMouseEvent<HTMLElement>) => {
+    e.preventDefault();
+    this.setState(s => ({ isOpen: !s.isOpen }));
+  };
+
+  render(): React.Node {
+    const { children, value, dropdownProps, ...buttonProps } = this.props;
+
+    const propsForDropdownMenu: DropdownMenuProps = dropdownProps
+      ? Object.assign(dropdownProps, { show: this.state.isOpen })
+      : {
+          show: this.state.isOpen,
+        };
+
+    const dropdownMenu = React.createElement(
```

Comment:
Curious as to why we are using React.createElement here as opposed to jsx?

Reply:
It's a workaround for playing nicer with flow, I tried `{...propsForDropdownMenu}` onto the jsx component but had inexact type errors.